src/System.Data.SqlClient/src/Microsoft/SqlServer/Server/MetadataUtilsSmi.cs

```diff
@@ -77,57 +73,60 @@ internal class MetaDataUtilsSmi
             SqlDbType.DateTimeOffset,       // System.DateTimeOffset
         };
 
-        // Hash table to map from clr type object to ExtendedClrTypeCodeMap enum
-        // this HashTable should only be accessed from DetermineExtendedTypeCode and class ctor for setup.
-        private static readonly Hashtable s_typeToExtendedTypeCodeMap = CreateTypeToExtendedTypeCodeMap();
+        // Dictionary to map from clr type object to ExtendedClrTypeCodeMap enum.
+        // This dictionary should only be accessed from DetermineExtendedTypeCode and class ctor for setup.
+        private static readonly Dictionary<Type, ExtendedClrTypeCode> s_typeToExtendedTypeCodeMap = CreateTypeToExtendedTypeCodeMap();
```

Comment:
Maybe in a future PR this could be replaced with a bunch of `if` statements, instead of initializing a `Dictionary` at runtime and filling it with types?